### ISSUE TITLE
SpaceAroundOperators raised when the previous aligned line has brackets

### ISSUE THREAD

**Issue by bekicot** [rubocop-hq/rubocop#4637]
Here is the example snippet to reproduce the error
```rb
mamamie = ''
test    = [
  1, 2, 3, 4, 5, 6, 7
]
hello   = [
  2, 2, 4, 5
]

mamamie + test[0].to_s + hello[2].to_s
```
--------

## Expected behavior

Rubocop should pass

## Actual behavior

rubocop raises an `Operator = should be surrounded by a single space.` error.

## Steps to reproduce the problem

run rubocop against
```rb
mamamie = ''
test    = [
  1, 2, 3, 4, 5, 6, 7
]
hello   = [
  2, 2, 4, 5
]

mamamie + test[0].to_s + hello[2].to_s
```
Or some other example
https://github.com/wikimedia/mediawiki-ruby-api/blob/master/mediawiki_api.gemspec

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by smakagon** [comment]
From what I see in specs to that Cop, it's expected behavior:
```ruby 
    it 'registers an offense for assignment with many spaces on either side' do
      expect_offense(<<-RUBY.strip_indent)
        x   = 0
            ^ Operator `=` should be surrounded by a single space.
        y +=   0
          ^^ Operator `+=` should be surrounded by a single space.
        z[0]  =  0
              ^ Operator `=` should be surrounded by a single space.
      RUBY
    end
```

Probably it's a good idea to be able to disable that check for assignment. 
But to fix this offense you can change code to:
```ruby
test = [
  1, 2, 3, 4, 5, 6, 7
]
```

**Comment by smakagon** [comment]
Sorry, you're right, those specs are for `AllowForAlignment: false` config. So looks like we have a bug here :)

**Comment by smakagon** [comment]
Here is my current findings:
In specs I see that `AllowForAlignment` is set to `true` but still specs expect this to add an offense:
`x   = 0`
I'm not sure if it's correct behavior with that config.

Speaking about this method:
```
        def excess_leading_space?(op, with_space)
          with_space.source =~ /^  / &&
            (!allow_for_alignment? || !aligned_with_operator?(op))
        end
```

For described case, this code `aligned_with_operator?(op)` returns `false` so this: `!aligned_with_operator?(op)` returns `true`. It makes this cop to log offense.

Seems like specs should be fixed and usage/logic of `aligned_with_operator?(op)` should be refined.

**Comment by jonas054** [comment]
My opinion is that it's a bug. The code in `preceding_following_alignment.rb` looks like we want to support the above case.

**Comment by stdedos** [comment]
Also seen in:

```ruby
OPEN_OBJECT = {
  'str1'      => {
    'str2'      => 'str3',
    'str4_long' => 'str5'
  },
  'str6'      => {},
  'str7_long' => { 'str8' => 'str9' }
}.freeze
```
